src/plugins/usage_collection/server/collector/collector.ts

```diff
@@ -42,41 +65,27 @@ export class Collector {
       throw new Error('Collector must be instantiated with a options.fetch function property');
     }
 
-    this.log = logger;
-
     Object.assign(this, options); // spread in other properties and mutate "this"
 
     this.type = type;
     this.init = init;
     this.fetch = fetch;
-
-    const defaultFormatterForBulkUpload = result => ({ type, payload: result });
-    this._formatForBulkUpload = formatForBulkUpload || defaultFormatterForBulkUpload;
-    if (typeof isReady === 'function') {
```

Comment:
Lets keep this `typeof` check. In case someone sets isReady to `true` the collectorSet will not throw when attempting to execute something other than a function.

Reply:
Sure! I was relying on TS to ensure users will only provide a function here, but I'll add the safety check just in case anyone does it in a `.js` or `as any` cast